src/engine/bank05.asm

```diff
@@ -4029,72 +4030,78 @@ Func_16695: ; 16695 (5:6695)
 	ld [wSelectedMoveIndex], a
 	call CheckEnergyNeededForAttack
 	jp c, .asm_1671e
-	ld a, $0c
+	ld a, $0c ; ATTACHED_ENERGY_BOOST
 	call CheckLoadedMoveFlag
 	jr c, .asm_166af
-	ld a, $0b
+	ld a, $0b ; FLAG_2_BIT_5
 	call CheckLoadedMoveFlag
 	jr c, .asm_16710
-	jp .asm_16775
+	jp .check_evolution
 
 .asm_166af
 	ld a, [wLoadedMoveUnknown1]
 	cp $02
 	jr z, .asm_166bc
 	call AddToAIScore
-	jp .asm_16775
+	jp .check_evolution
 
 .asm_166bc
 	call Func_171fb
 	jr c, .asm_166cd
-	cp $03
+	cp 3
 	jr c, .asm_166cd
 .asm_166c5
 	ld a, 5
 	call SubFromAIScore
-	jp .asm_16775
+	jp .check_evolution
 .asm_166cd
 	ld a, 2
 	call AddToAIScore
 
+; check whether move has ATTACHED_ENERGY_BOOST flag
+; and add to AI score if attaching another energy
+; will KO defending Pokémon
 	ld a, $0c
 	call CheckLoadedMoveFlag
-	jp nc, .asm_16775
+	jp nc, .check_evolution
 	ld a, [wSelectedMoveIndex]
 	call CalculateMoveDamage_VersusDefendingCard
 	ld a, DUELVARS_ARENA_CARD_HP
 	call GetNonTurnDuelistVariable
 	ld hl, wDamage
 	sub [hl]
-	jp c, .asm_16775
-	jp z, .asm_16775
+	jp c, .check_evolution
+	jp z, .check_evolution
 	ld a, [wDamage]
-	add 10
+	add 10 ; boost gained by attaching another energy card
 	ld b, a
 	ld a, DUELVARS_ARENA_CARD_HP
 	call GetNonTurnDuelistVariable
 	sub b
 	jr c, .asm_166ff
-	jr nz, .asm_16775
+	jr nz, .check_evolution
 .asm_166ff
 	ld a, 20
 	call AddToAIScore
 	ldh a, [hTempPlayAreaLocation_ff9d]
 	or a
-	jr nz, .asm_16775
+	jr nz, .check_evolution
 	ld a, 10
 	call AddToAIScore
-	jr .asm_16775
+	jr .check_evolution
 
+; FLAG_2_BIT_5 is set only for Pokémon Powers,
+; except for Magnemite2's Magnetic Storm attack
 .asm_16710
 	ld a, [wLoadedCard1ID]
 	cp ZAPDOS2
-	jr z, .asm_16775
+	jr z, .check_evolution
 	call Func_171fb
 	jr c, .asm_166cd
 	jr .asm_166c5
+
 .asm_1671e
-	ld a, $0d
+	ld a, $0d ; FLAG_2_BIT_5
```

Comment:
For now, I would use for these:
``$1 <<  3 | FLAG_2_BIT_5``

Using for the address bits ``$0``, ``$1``, or ``$2`` and using the corresponding flag constant.

Reply:
That's actually not right, since FLAG_2_BIT_5 is %00100000, so that results in %00101000. It would work if FLAG_2_BIT_5 was defined by a bit shift of some constant (EQU 5), but as of now that is not the case.

Reply:
True, my bad. Leave them hardcoded for now and I'll take care of those myself at some point.

Reply:
Is there any good reason to keep using `$1` and friends (instead of just `1`)? Those were an artifact of the disassembly in the mainstream gen 1/2 repos; not sure of why you'd bring more of those in.

Reply:
I like ``$1`` to signal that it's something that should be a constant that hasn't been added or figured out yet. Otherwise I agree that for arithmetic operations and the like I also prefer ``1``.